### PR TITLE
Issue 4782: Make LocalPravegaEmulator usable by downstream applications

### LINKED ISSUE #4782: Best practices for Integration Tests on a Java Client application
I am looking for some suggestion about creating integration (unit) tests for a client application written in Java.

Ideally I would like to start an in memory Standalone Pravega service.

```
 try (PravegaStandaloneForTests server = bootPravegaStandaloneForTests() {   
  String  connectionString = server.getConnectionString();
   run application that writes/reads to Pravega....
}

```

Context:
We are going to implement an integration between Pravega and [EmailSuccess.com](https://emailsuccess.com), EmailSuccess.com will be a source of data, it will write a stream of events to Pravega.
We want to create integration  tests with JUnit.

### PR DESCRIPTION
**Pull request by eolivelli**
**Change log description**  
- make the 'start' method public
- add a meaningful javadoc that explains that this API is intended to be used from applications that want to run tests against a real Pravega cluster

**Purpose of the change**  
Fixes #4782

**What the code does**  
- make the 'start' method public
- add Javadoc

**How to verify it**  
Test should pass.
A simple application like https://github.com/eolivelli/pravega-testing should work without using reflection in order to start the Emulator

### CODE REVIEW

**Comment by fpj** [comment]
@eolivelli you missed the sign-off, you may want to amend the commit and force push.

**Comment by eolivelli** [comment]
@fpj sure. done. thanks for  your heads up

**Comment by eolivelli** [comment]
tests are failing, I am checking

**Comment by eolivelli** [comment]
I have simplified the patch.
The initial intent was not to start ZK server at all, but it is not strictly needed and it is messing up tests.
Let's start with the minimal change that achieves the goal.

If you will cut release a 0.7.1 please backport this fix to that version.

File: standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java [hunk 1]
```diff
@@ -14,6 +14,12 @@
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
+/**
+ * Main entry point for Integration tests that need an in process Pravega cluster.
+ * This class is intended to be used both by internal test suites
+ * and  by test suites of applications that want to run tests against
```
Comment (ravisharda) [inline standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java:20]: nit: an extra space before `by` and `in-process`

**Comment by ravisharda** [comment]
@eolivelli the code coverage check's requirements haven't been met. Could you please add unit tests for your changes?

**Comment by eolivelli** [comment]
I have only added a javadoc and made 'public' an existing method.

What should I do?

File: standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java (hunk 1 above)
Comment (eolivelli) [reply standalone/src/main/java/io/pravega/local/LocalPravegaEmulator.java:20]: @ravisharda I have corrected the extra space and the in-process word.
I can't see where to add another '.'

**Comment by andreipaduroiu** [comment]
@fpj we need some help with codecov here. Same issue as in other PRs we have.

**Comment by eolivelli** [comment]
Thank you.
I will update my example as soon as 0.8.0 will be released.
Or perhaps 0.7.1 of you cherry pick

**Pull request merged**